### ISSUE TITLE
Dynamics 365 CE API Backup to Azure not working

### ISSUE THREAD

**Issue by stephenacrown** [WaelHamze/dyn365-ce-vsts-tasks#37]
Hi,

I cannot get Dynamics 365 API Backup to Azure working as per error below:

![image](https://user-images.githubusercontent.com/38353501/50995579-66860100-1517-11e9-9fd6-d596ecd4f3bb.png)

This is documented Dynamics 365 CE API shows this as a supported target destination, and is also included in the tasks in this marketplace extension (again presumably because it is documented!)

https://docs.microsoft.com/en-gb/rest/api/admin.services.crm.dynamics.com/instancebackups/backupinstance#azurestorage

Has anyone else got this working or using it?

If not, are there any thoughts on workarounds for backups to somewhere else (to protect against an instance/environment deletion which takes all backups along with it for example)?

I have raised this with Microsoft as well, and will post back here any findings.

**Comment by stephenacrown** [comment]
Thanks for the heads up. 👍

I didn’t note this subtlety in the documentation.....

I think it would be useful if the task had this removed or at least indicated this is not currently supported?

I’ll follow this up with Microsoft to see what they say, what alternatives, mitigation’s they have, and also see if they’ll tell me if it’s coming back. 

I’ll post here any further updates.

**Comment by WaelHamze** [comment]
@stephenacrown any updates from your conversations with MS?

**Comment by stephenacrown** [comment]
Hi @waelhamze

Yes, a couple responses as follows:



From Microsoft support:

“I have researched on the issue you mentioned and understood that Perform backups to Azure storage is not supported and is only available for Microsoft Internal usage.
 
Request you to go through the below links/details which would help you understand our considerations and get further clarification on the ask:
 
Considerations:
Operation Management API: https://docs.microsoft.com/en-us/dynamics365/customer-engagement/developer/online-management-api/operations-supported#back-up-and-restore-instances
For Version 9
It is possible to retrieve a list of instance backups through API
It is possible to backup an instance through API
It is possible to restore an instance through API
If you access the backup instance through API section you get directed to: https://docs.microsoft.com/pt-pt/rest/api/admin.services.crm.dynamics.com/instancebackups/backupinstance
It shows on the request body: AzureStorageInformation as of Azure Storage type.
If you scroll further down you’ll get this https://docs.microsoft.com/pt-pt/rest/api/admin.services.crm.dynamics.com/instancebackups/backupinstance#azurestorage :
 
AzureStorage
Name

Type

Description

ContainerName

·   string

For internal use only.
StorageAccountKey
·   string

For internal use only.
StorageAccountName
·   string

For internal use only.
 
 
Conclusion: Perform backups to Azure Storage is not supported, and is only available for Microsoft Internal usage.
 
Alternative Recommendations:
Use the Data Export Service add-on from Appsource: https://docs.microsoft.com/en-us/dynamics365/customer-engagement/admin/replicate-data-microsoft-azure-sql-database
Looks for third party tools and partners that may be able to deliver capabilities not currently available in the product”


I responded and thanked them, and then queried whether they (Microsoft support) had the ability to restore an environment/instance that had been deleted (for example via accidental or malicious means), and they responded:


“As asked below, yes we can restore the deleted instance. Having that said, we can only restore the instance with in 3-4 weeks of deletion.”

I have responded back to this asking for SLA or similar based documentation for this statement. 

Hope that helps some others, but doesn’t look like the “backup to Azure” feature is coming back that I can see. 

Stephen
